blockstore/apps/search/tests/test_documents.py

```diff
@@ -0,0 +1,43 @@
+""" Tests for views. """
+
+from elasticsearch_dsl import Object
+
+from ..core.serializers import FieldSetSerializer
+from ..core.tests.test_base import DocumentBaseTestCase
+from ..documents import BlockDocument
+from ..fieldsets import (
+    AnalyticsFieldSet,
+    AuthorshipFieldSet,
+    EntityFieldSet,
+    OwnernershipFieldSet,
+    SummaryFieldSet,
+    TagsFieldSet,
+)
+
+
+class BlockDocumentTestCase(DocumentBaseTestCase):
+    """ Tests for BlockDocument. """
+
+    document_class = BlockDocument
+
+    def test_fieldsets_initialization(self):
+
+        for field_set_class in BlockDocument.FIELD_SETS:
+
+            document_field = BlockDocument._doc_type.mapping[field_set_class.name]  # pylint: disable=protected-access
+            assert isinstance(document_field, Object)
+
+            document_serializer = BlockDocument.Serializer()
+            assert isinstance(document_serializer.fields[field_set_class.name], FieldSetSerializer)
+
+    def test_document_create(self):
+
+        document = BlockDocument(
+            analytics=AnalyticsFieldSet.Document(favorites=8, remixes=2, views=200),
+            authorship=AuthorshipFieldSet.Document(author_ids=['First author', 'Second author']),
+            entity=EntityFieldSet.Document(type='html', id='html1'),
```

Comment:
A more realistic entity ID would be `type='xblock', id='block-v1:...'`

What exactly is in this index, btw? Is it just XBlocks, or is it XBlocks, courses, pathways, and libraries? If it's just XBlocks, we only need the UsageKey and the `type` is not part of the ID. But if we're mixing XBlocks, pathways, etc. then we might need the `type` and the `id` which together comprise the compound key.

Reply:
Yes, the index can contain documents for the metadata/searchable content for any kind of entity we want searchable. It can be XBlocks, courses, pathways, even pdf files, images etc (as long as the fields are correctly filled in). And yes the `(type, id)` is acting as the compound key.

Reply:
I have updated the example to `entity=EntityFieldSet.Document(type='xblock', id='block-v1:Blockstore+Course+type@html+block@030e35c'),`